src/libraries/System.Private.CoreLib/src/System/TimeSpan.cs

```diff
@@ -545,50 +522,65 @@ public static bool TryParseExact([NotNullWhen(true)] string? input, [NotNullWhen
-        {
-            return TimeSpanFormat.FormatC(this);
-        }
-        public string ToString([StringSyntax(StringSyntaxAttribute.TimeSpanFormat)] string? format)
-        {
-            return TimeSpanFormat.Format(this, format, null);
-        }
-        public string ToString([StringSyntax(StringSyntaxAttribute.TimeSpanFormat)] string? format, IFormatProvider? formatProvider)
-        {
-            return TimeSpanFormat.Format(this, format, formatProvider);
-        }
+        public override string ToString() => TimeSpanFormat.FormatC(this);
+        public string ToString([StringSyntax(StringSyntaxAttribute.TimeSpanFormat)] string? format) => TimeSpanFormat.Format(this, format, null);
+        public string ToString([StringSyntax(StringSyntaxAttribute.TimeSpanFormat)] string? format, IFormatProvider? formatProvider) => TimeSpanFormat.Format(this, format, formatProvider);
 
-        public bool TryFormat(Span<char> destination, out int charsWritten, [StringSyntax(StringSyntaxAttribute.TimeSpanFormat)] ReadOnlySpan<char> format = default, IFormatProvider? formatProvider = null) =>
-            TimeSpanFormat.TryFormat(this, destination, out charsWritten, format, formatProvider);
+        public bool TryFormat(Span<char> destination, out int charsWritten, [StringSyntax(StringSyntaxAttribute.TimeSpanFormat)] ReadOnlySpan<char> format = default, IFormatProvider? formatProvider = null)
+            => TimeSpanFormat.TryFormat(this, destination, out charsWritten, format, formatProvider);
 
         /// <inheritdoc cref="IUtf8SpanFormattable.TryFormat" />
-        public bool TryFormat(Span<byte> utf8Destination, out int bytesWritten, [StringSyntax(StringSyntaxAttribute.TimeSpanFormat)] ReadOnlySpan<char> format = default, IFormatProvider? formatProvider = null) =>
-            TimeSpanFormat.TryFormat(this, utf8Destination, out bytesWritten, format, formatProvider);
+        public bool TryFormat(Span<byte> utf8Destination, out int bytesWritten, [StringSyntax(StringSyntaxAttribute.TimeSpanFormat)] ReadOnlySpan<char> format = default, IFormatProvider? formatProvider = null)
+            => TimeSpanFormat.TryFormat(this, utf8Destination, out bytesWritten, format, formatProvider);
 
         #endregion
 
         public static TimeSpan operator -(TimeSpan t)
         {
-            if (t._ticks == MinValue._ticks)
-                throw new OverflowException(SR.Overflow_NegateTwosCompNum);
+            if (t._ticks == MinTicks)
+            {
+                ThrowHelper.ThrowOverflowException_NegateTwosCompNum();
+            }
             return new TimeSpan(-t._ticks);
         }
 
-        public static TimeSpan operator -(TimeSpan t1, TimeSpan t2) => t1.Subtract(t2);
+        public static TimeSpan operator -(TimeSpan t1, TimeSpan t2)
+        {
+            long result = t1._ticks - t2._ticks;
+
+            if (((t1._ticks >> 63) != (t2._ticks >> 63)) && ((t1._ticks >> 63) != (result >> 63)))
```

Comment:
>(t1._ticks >> 63 [](http://example.com/codeflow?start=17&length=16)

does the jit optimize this calculation as it is used ywice in the statement? or should we store it and then use it?

Reply:
It does. We can manually cache it and simplify things, however.